File: 윤상우/Week_5/P42840_모의고사.java
package temp;

import java.util.ArrayList;
import java.util.List;

class Solution {

    static int [] sol1 ={1,2,3,4,5};
    static int [] sol2 ={2,1,2,3,2,4,2,5};
    static int [] sol3 ={3,3,1,1,2,2,4,4,5,5};

    public int[] solution(int[] answers) {
        int len = answers.length;

        int [] p = new int [4];

        for(int i=0; i<len; i++){
            if(answers[i] == sol1[i%5]) p[1]++;
            if(answers[i] == sol2[i%8]) p[2]++;
            if(answers[i] == sol3[i%10]) p[3]++;
        }

        int max = Math.max(p[1], Math.max(p[2], p[3]));

        List<Integer> answerList = new ArrayList<>();

        for(int i=1; i<4; i++){
            if(p[i] == max) answerList.add(i);
        }
        int size = answerList.size();
        int [] answer = new int [size];

        for(int i=0; i<size; i++){
            answer[i] = answerList.get(i);
        }

        return answer;
    }
}
public class P42840_모의고사 {
    public static void main(String[] args) {

    }
}

File: 신선영/Week_5/P42840_모의고사.py
def solution(answers):
    s1 = [1, 2, 3, 4, 5]
    s2 = [2, 1, 2, 3, 2, 4, 2, 5]
    s3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    answer = []
    idx1, idx2, idx3 = 0, 0, 0
    cnt = [0, 0, 0] # 각 학생별 맞은 개수 저장
    
    for a in answers:
        # 만약 답과 학생의 현재 찍은 값이 갔다면 맞은 개수 추가
        if s1[idx1] == a:
            cnt[0] += 1
        if s2[idx2] == a:
            cnt[1] += 1
        if s3[idx3] == a:
            cnt[2] += 1
        idx1 += 1
        idx2 += 1
        idx3 += 1
        # 한바퀴를 다 돌았을 경우 초기화
        if idx1 == len(s1):
            idx1 = 0
        if idx2 == len(s2):
            idx2 = 0
        if idx3 == len(s3):
            idx3 = 0
        
    m = max(cnt)
    # 가장 많은 개수를 맞힌 학생 모두 answer에 추가
    for i in range(3):
        if cnt[i] == m:
            answer.append(i + 1)
    return answer